### PR TITLE
support cast and change the name of expression select field.

### PR DESCRIPTION
**Pull request by hanfei19910905**
@zimulala @shenli @coocood

### CODE REVIEW

**Comment by coocood** [comment]
add test case for cast.

**Comment by coocood** [comment]
LGTM

File: evaluator/builtin_other.go [hunk 1]
```diff
@@ -390,3 +390,19 @@ func unaryOpFactory(op opcode.Op) BuiltinFunc {
 		return
 	}
 }
+
+// CastFuncFactory produces builtin function according to field types.
+func CastFuncFactory(tp *types.FieldType) (BuiltinFunc, error) {
+	switch tp.Tp {
+	case mysql.TypeString, mysql.TypeDuration, mysql.TypeDatetime,
```
Comment (shenli) [inline evaluator/builtin_other.go]: Add comments for this check.
It is better to add the refer: http://dev.mysql.com/doc/refman/5.7/en/cast-functions.html

File: plan/newplanbuilder.go [hunk 2]
```diff
@@ -262,11 +263,11 @@ func (b *planBuilder) buildProjection(p LogicalPlan, fields []*ast.SelectField,
 		var tblName, colName model.CIStr
 		if field.AsName.L != "" {
 			colName = field.AsName
-		} else if c, ok := newExpr.(*expression.Column); ok {
+		} else if c, ok := newExpr.(*expression.Column); ok && !c.IsAgg {
 			colName = c.ColName
 			tblName = c.TblName
 		} else {
-			colName = model.NewCIStr(field.Expr.Text())
+			colName = model.NewCIStr(field.Text())
```
Comment (shenli) [inline plan/newplanbuilder.go]: Any test to cover this?

Comment (hanfei19910905) [inline plan/newplanbuilder.go]: the test framework seems can't check column name.

File: evaluator/builtin_other.go [hunk 3]
```diff
@@ -392,6 +392,7 @@ func unaryOpFactory(op opcode.Op) BuiltinFunc {
 }
 
 // CastFuncFactory produces builtin function according to field types.
+// See https://dev.mysql.com/doc/refman/5.7/en/cast-functions.html
```
Comment (shenli) [inline evaluator/builtin_other.go]: See:

**Comment by shenli** [comment]
LGTM

**Pull request merged**